client/rpcserver/rpcserver_test.go

```diff
@@ -499,7 +494,7 @@ func TestClientMap(t *testing.T) {
 	conn := new(TConn)
 
 	go s.websocketHandler(conn, "someip")
-	time.Sleep(time.Millisecond)
+	time.Sleep(time.Millisecond * 5)
```

Comment:
OK, but with new tests, we should consider carefully if a Sleep is always going to do the trick.  Of course we need to weight the burden of proper synchronization that only plays a role in the test setup, but given how often these fail, esp. on CI, we'll probably need to do it properly more often in the future.

Reply:
I agree and take full responsibility for the time this pattern has already cost us. It seems a little tedious, but something like

```
tries := 0
for {
	tries++
	s.mtx.RLock()
	num := len(s.clients)
	s.mtx.RUnlock()
	if num == 1 {
		break
	}
	if tries > 100 {
		t.Fatalf("failed to start websocket handler")
	}
	time.Sleep(time.Millisecond)
}
```
should work.

Reply:
Sure, a waitforclient kind of function should be helpful.
In other cases I wonder if exposing some kind of signal or checkable value might be helpful, although I don't like the idea of modifying the actual product code just for tests.

Reply:
I modified the product code...